### PR TITLE
When uploading a new FxOS Add-on version, make older pending versions obsolete (bug 1203594)

### PR DESCRIPTION
**Pull request by diox**
https://bugzilla.mozilla.org/show_bug.cgi?id=1203594

### CODE REVIEW

File: mkt/extensions/tests/test_models.py [hunk 1]
```diff
@@ -527,3 +562,19 @@ def test_pending(self):
         Extension.objects.create(status=STATUS_NULL)
 
         eq_(list(Extension.objects.pending()), [extension1, extension2])
+
+
+class TestExtensionVersionManager(TestCase):
+    def test_pending(self):
```
Comment (chuckharmston) [inline mkt/extensions/tests/test_models.py]: Probably also want a test for `Extension.objects.public`.

File: mkt/extensions/tests/test_models.py [hunk 2]
```diff
@@ -527,3 +562,19 @@ def test_pending(self):
...
+            extension=extension1, status=STATUS_PENDING, version='1.1')
+        extension2 = Extension.objects.create()
+        version2 = ExtensionVersion.objects.create(
+            extension=extension2, status=STATUS_PENDING, version='2.1')
+        ExtensionVersion.objects.create(
+            extension=extension2, status=STATUS_PUBLIC, version='2.2')
+        Extension.objects.create(status=STATUS_PUBLIC)
+        Extension.objects.create(status=STATUS_NULL)
+
+        eq_(list(ExtensionVersion.objects.pending()), [version1, version2])
```
Comment (chuckharmston) [inline mkt/extensions/tests/test_models.py]: This test is sort of brittle. I guess it gets the job done, but perhaps it's less brittle if you do something like `[ev for ev in ExtensionVersion.objects.all() if ev.status == STATUS_PENDING]`?

File: mkt/extensions/models.py [hunk 3]
```diff
@@ -285,6 +302,22 @@ def remove_signed_file(self):
...
 
+    def set_older_pending_versions_as_obsolete(self):
+        """Set all pending versions older than this one attached to the same
+        Extension as DISABLED (obsolete).
+
+        To be on the safe side this method does not trigger signals and needs
+        to be called when creating a new pending version, before actually
+        changing its status to PENDING. That way we avoid having extra versions
+        laying around when we automatically update the status on the parent
+        Extension."""
```
Comment (chuckharmston) [inline mkt/extensions/models.py]: Took me a few reads to get my head around it, but this is really smart.

**Comment by chuckharmston** [comment]
r+wc, does what it says. Nice work.

File: mkt/extensions/tests/test_models.py (hunk 2 above)
Comment (diox) [inline mkt/extensions/tests/test_models.py]: Yeah, but doing that I would almost replicate the code in the manager method itself... 

I re-organized it a little more to create fewer objects, that should make it easier on the eyes at least.

File: mkt/extensions/tests/test_models.py (hunk 1 above)
Comment (diox) [inline mkt/extensions/tests/test_models.py]: Done

**Comment by diox** [comment]
Closing to re-open a new PR with the right branch.

**Pull request closed without merging**